From c0b5cf9e40eeaf58b7007583ed08e2db52d257c7 Mon Sep 17 00:00:00 2001
From: Konrad Kleine <kkleine@redhat.com>
Date: Wed, 20 Oct 2021 18:12:24 +0200
Subject: [PATCH 3/3] Help compiler-rt find shared LLVM cmake files from
 llvm-cmake package

Consume top-level cmake/Modules/* files from /usr/lib64/cmake/llvm or /usr/lib/cmake/llvm
---
 compiler-rt/CMakeLists.txt              | 2 ++
 compiler-rt/lib/builtins/CMakeLists.txt | 5 ++++-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/compiler-rt/CMakeLists.txt b/compiler-rt/CMakeLists.txt
index 17d7e4ffbdeb..4ce3a2c78442 100644
--- a/compiler-rt/CMakeLists.txt
+++ b/compiler-rt/CMakeLists.txt
@@ -12,12 +12,14 @@ if (CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR OR COMPILER_RT_STANDALONE
   set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 endif()
 
+include(GNUInstallDirs)
 # Add path for custom compiler-rt modules.
 list(INSERT CMAKE_MODULE_PATH 0
   "${CMAKE_CURRENT_SOURCE_DIR}/cmake"
   "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules"
   "${CMAKE_CURRENT_SOURCE_DIR}/../cmake"
   "${CMAKE_CURRENT_SOURCE_DIR}/../cmake/Modules"
+  "/usr/${CMAKE_INSTALL_FULL_LIBDIR}/cmake/llvm"
   )
 
 if(CMAKE_CONFIGURATION_TYPES)
diff --git a/compiler-rt/lib/builtins/CMakeLists.txt b/compiler-rt/lib/builtins/CMakeLists.txt
index 971e94407a2c..3391f4ad1b57 100644
--- a/compiler-rt/lib/builtins/CMakeLists.txt
+++ b/compiler-rt/lib/builtins/CMakeLists.txt
@@ -9,11 +9,14 @@ if (CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
   project(CompilerRTBuiltins C ASM)
   set(COMPILER_RT_STANDALONE_BUILD TRUE)
   set(COMPILER_RT_BUILTINS_STANDALONE_BUILD TRUE)
+  include(GNUInstallDirs)
   list(INSERT CMAKE_MODULE_PATH 0
     "${CMAKE_SOURCE_DIR}/../../cmake"
     "${CMAKE_SOURCE_DIR}/../../cmake/Modules"
     "${CMAKE_SOURCE_DIR}/../../../cmake"
-    "${CMAKE_SOURCE_DIR}/../../../cmake/Modules")
+    "${CMAKE_SOURCE_DIR}/../../../cmake/Modules"
+    "/usr/${CMAKE_INSTALL_FULL_LIBDIR}/cmake/llvm"
+  )
   include(base-config-ix)
   include(CompilerRTUtils)
 
-- 
2.31.1

